adhocracy4/follows/static/follows/FollowButton.jsx

```diff
@@ -1,77 +1,77 @@
+import django from 'django'
+import React, { useState, useEffect } from 'react'
+import Alert from '../../../static/Alert'
+
 const api = require('../../../static/api')
-const django = require('django')
-const React = require('react')
-const Alert = require('../../../static/Alert')
 
-class FollowButton extends React.Component {
-  constructor (props) {
-    super(props)
+const translated = {
+  followDescription: django.gettext('Click to be updated about this project via email.'),
+  followingDescription: django.gettext('Click to no longer be updated about this project via email.'),
+  followAlert: django.gettext('You will be updated via email.'),
+  followingAlert: django.gettext('You will no longer be updated via email.'),
+  follow: django.gettext('Follow'),
+  following: django.gettext('Following')
+}
 
-    this.state = {
-      followed: undefined,
-      follows: 0,
-      alert: null
-    }
-  }
+export const FollowButton = (props) => {
+  const [followed, setFollowed] = useState(undefined)
```

Comment:
so what I thought was that on page load the state would be undefined as it does not yet know if the user is following or not and after the useEffect then the state can be set correctly. Thinking about this further the button should probably be set to disabled too while it is updated, this is relevant for users on slow mobile data to avoid double clicks, however @goapunk just sent me this about the pressed state and as we do need to update the wording i'm back to being unsure :p
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-pressed